Feature/11355 fix profile button in altinn header

WilliamThorenfeldt opened the pull request:
## Description
- Fixing Profile Menu in Altinn Header to be a a button so it gets the correct behaviour. 

## Related Issue(s)
- #11355 

## Video
https://github.com/Altinn/altinn-studio/assets/133344438/cd3989d8-2679-4ed4-8730-e5b36088a3c4

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] Cypress tests run green locally (https://github.com/Altinn/altinn-studio/tree/master/frontend/testing/cypress#run-altinn-studio-tests)

JamalAlabdullah reviewed (changes requested):
I just tried to test before the review, and I found some issues:
-  Don't we need a title when we hover, as described in the issue?
-  After clicking on the icon, the popover keeps showing. It does not disappear even when we click outside (anywhere on the page).
-   I'm not sure, but should we have a background when we hover?
-   Maybe we need this implementation on the dashboard as well.

WilliamThorenfeldt commented:
> I just tried to test before the review, and I found some issues :
> 
> * Don't we need a title when we hover, as described in the issue?
> * After clicking on the icon, the popover keeps showing. It does not disappear even when we click outside (anywhere on the page).
> * I'm not sure, but should we have a background when we hover? 🙂
> * Maybe we need this implementation on the dashboard as well? 🙂

Thank you for pointing out. 
- Title is now added as described in the issue. 
- Nice spotted that it does not close. This is now fixed. 
- In terms of hovering, we are now using the button from Designsystemet, so hover comes from there. 
- I think it is a good idea to add this header to the dashboard as well. I can see that the headers used for dashboard is different from the one in app-development. However, that is for another PR and issue 😄 

Updated video showing new changes 😄 
https://github.com/Altinn/altinn-studio/assets/133344438/c2801f5c-0d65-4724-a311-4ee5681ec42b

JamalAlabdullah reviewed (approved):
LGFM 👍
